response_operations_ui/views/accounts.py

```diff
@@ -0,0 +1,204 @@
+
+    return redirect(url_for('account_bp.request_new_account_check_email', email=url_safe_serializer.dumps(email)))
+
+
+@account_bp.route('/resend_account_email_expired_token/<token>', methods=['GET'])
+def resend_account_email_expired_token(token):
+    email = token_decoder.decode_email_token(token)
+    return send_create_account_email(email)
+
+
+@account_bp.route('/create-account/<token>', methods=['GET'])
+def get_create_account(token, form_errors=None):
+    form = CreateAccountForm(request.form)
+
+    try:
+        duration = app.config['EMAIL_TOKEN_EXPIRY']
+        _ = token_decoder.decode_email_token(token, duration)
+    except SignatureExpired:
+        logger.warning('Token expired for Response Operations account creation', token=token)
+        return render_template('request-new-account-expired.html', token=token)
+    except (BadSignature, BadData):
+        logger.warning('Invalid token sent to Response Operations account creation', token=token)
+        return render_template('request-new-account-expired.html', token=token)
+
+    template_data = {
+        "error": {
+            "type": form_errors
+        },
+        'token': token
+    }
+    return render_template('create-new-account.html', form=form, data=template_data)
+
+
+@account_bp.route('/create-account/<token>', methods=['POST'])
+def post_create_account(token):
+    form = CreateAccountForm(request.form)
+
+    if not form.validate():
+        return get_create_account(token, form_errors=form.errors)
+    
+    try:
+        duration = app.config['EMAIL_TOKEN_EXPIRY']
+        email = token_decoder.decode_email_token(token, duration)
+    except SignatureExpired:
+        logger.warning('Token expired for Response Operations create account', token=token)
+        return render_template('request-new-account-expired.html', token=token)
+    except (BadSignature, BadData):
+        logger.warning('Invalid token sent to Response Operations create account', token=token)
+        return render_template('request-new-account-expired.html', token=token)
+    
+    password = request.form.get('password')
+    user_name = request.form.get('user_name')
+    first_name = request.form.get('first_name')
+    last_name = request.form.get('last_name')
+
+    response = uaa_controller.create_user_account(email, password, user_name, first_name, last_name)
+
+    if response is not None:
+        if response.status_code == 201:
+            logger.info('Successfully created user account', token=token)
+            send_confirm_created_email(email, first_name)
+            flash('Account successfully created', category='account_created')
+            return redirect(url_for('sign_in_bp.sign_in'))
+        if response.status_code == 409:
+            # Username already exists
+            form_errors = {'user_name': ["Username already in use; please choose another"]}
```

Comment:
It's probably not an issue because it's an internal system, but we're okay giving the user information about what's valid and what isn't?  I ask because it's not something we'd do for an externally facing site as we'd want to give something probing the site as little info as possible.

Reply:
I disagree - this is absolutely information we'd need to give on an externally-facing site if someone chose a duplicate username - the process can't continue with a duplicate and we need to inform the user as to why. Avoiding site probing is always an aim but not always achievable.

Reply:
On reflection this is kinda obvious.  We can't just say 'no account... no we're not going to tell you why'